### PR TITLE
Fix bigsur benchmark build

### PR DESCRIPTION
**Pull request by 2bethere**
### Description
On MacOS BigSur Java 11 builds, maven-assembly-plugin used under benchmark module throws an error.
`[ERROR] Failed to execute goal org.apache.maven.plugins:maven-assembly-plugin:2.6:single (default) on project druid-benchmarks: Execution default of goal org.apache.maven.plugins:maven-assembly-plugin:2.6:single failed: An API incompatibility was encountered while executing org.apache.maven.plugins:maven-assembly-plugin:2.6:single: java.lang.ExceptionInInitializerError: null`

Fixed the bug by upgrading maven-assembly-plugin to the latest (3.2.0)

Also made a fix to license generation python scripts where newer version of pyyaml's load_all requires Loader to be specified. 

This PR has:
- [x] been self-reviewed.
- [ ] using the [concurrency checklist](https://github.com/apache/druid/blob/master/dev/code-review/concurrency.md) (Remove this item if the PR doesn't have any relation to concurrency.)
- [ ] added documentation for new or modified features or behaviors.
- [ ] added Javadocs for most classes and all non-trivial methods. Linked related entities via Javadoc links.
- [ ] added or updated version, license, or notice information in [licenses.yaml](https://github.com/apache/druid/blob/master/dev/license.md)
- [ ] added comments explaining the "why" and the intent of the code wherever would not be obvious for an unfamiliar reader.
- [ ] added unit tests or modified existing tests to cover new code paths, ensuring the threshold for [code coverage](https://github.com/apache/druid/blob/master/dev/code-review/code-coverage.md) is met.
- [ ] added integration tests.
- [ ] been tested in a test Druid cluster.

### CODE REVIEW

**Comment by FrankChen021** [comment]
One of my project once suffered this problem and the solution is update the plugin to 3.3.0. So I see this change is reasonable.

File: benchmarks/pom.xml [hunk 1]
```diff
@@ -200,23 +200,25 @@
       <plugin>
         <groupId>org.apache.maven.plugins</groupId>
         <artifactId>maven-assembly-plugin</artifactId>
-        <version>2.6</version>
+        <version>3.2.0</version>
```
Comment (FrankChen021) [inline benchmarks/pom.xml:203]: the version is 3.2.0 not the 3.3.0 you mentioned in the description of this PR. which one should it be?

Comment (2bethere) [reply benchmarks/pom.xml:203]: It is 3.2.0. The commit message was a typo.

**Review by FrankChen021 — APPROVED** [review]

**Comment by 2bethere** [comment]
Going to open a new one on latest version.

**Pull request closed without merging**